deepchem/data/data_loader.py

```diff
@@ -368,45 +291,91 @@ class CSVLoader(DataLoader):
   pandas, but this class may prove useful if you're processing
   large CSV files that you don't want to manipulate directly in
   memory.
+
+  Examples
+  --------
+  Let's suppose we have some smiles and labels
+
+  >>> smiles = ["C", "CCC"]
+  >>> labels = [1.5, 2.3]
+
+  Let's put these in a dataframe.
+
+  >>> import pandas as pd
+  >>> df = pd.DataFrame(list(zip(smiles, labels)), columns=["smiles", "task1"])
+
+  Let's now write this to disk somewhere. We can now use `CSVLoader` to
+  process this CSV dataset.
+
+  >>> import tempfile
+  >>> import deepchem as dc
+  >>> with tempfile.NamedTemporaryFile(mode='w') as tmpfile:
+  ...   df.to_csv(tmpfile.name)
+  ...   loader = dc.data.CSVLoader(["task1"], feature_field="smiles",
+  ...                              featurizer=dc.feat.CircularFingerprint())
+  ...   dataset = loader.create_dataset(tmpfile.name)
+  >>> len(dataset)
+  2
+
+  Of course in practice you should already have your data in a CSV file if
+  you're using `CSVLoader`. If your data is already in memory, use
+  `InMemoryLoader` instead.
   """
 
   def __init__(self,
-               tasks,
-               smiles_field=None,
+               tasks: OneOrMany[str],
+               feature_field: Optional[str] = None,
+               label_field: Optional[str] = None,
+               weight_field: Optional[str] = None,
+               smiles_field: Optional[str] = None,
                id_field=None,
-               featurizer=None,
+               featurizer: Optional[Featurizer] = None,
                log_every_n=1000):
     """Initializes CSVLoader.
 
     Parameters
     ----------
-    tasks: list[str]
+    tasks : List[str]
       List of task names
-    smiles_field: str, optional
-      Name of field that holds smiles string
-    id_field: str, optional
-      Name of field that holds sample identifier
+    feature_field : str, optional (default None)
+      Field with data to be featurized.
+    id_field: str, optional, (default None)
+      CSV column that holds sample identifier
+    smiles_field: str, optional (DEPRECATED)
+      Name of field that holds smiles string 
     featurizer: dc.feat.Featurizer, optional
       Featurizer to use to process data
     log_every_n: int, optional
       Writes a logging statement this often.
     """
     if not isinstance(tasks, list):
       raise ValueError("tasks must be a list.")
+    if smiles_field is not None:
+      logger.warning(
+          "smiles_field is deprecated and will be removed in a future version of DeepChem. Use feature_field instead."
+      )
+      if feature_field is not None and smiles_field != feature_field:
+        raise ValueError(
+            "smiles_field and feature_field if both set must have the same value."
+        )
+      elif feature_field is None:
+        feature_field = smiles_field
+
     self.tasks = tasks
-    self.smiles_field = smiles_field
+    self.feature_field = feature_field
+    self.id_field = id_field
```

Comment:
This line is necessary...?

Reply:
I think we use `self.id_field` later down so I think so. I might be misunderstanding your comment though!